libdevcrypto/SecretStore.cpp

```diff
@@ -183,62 +190,66 @@ h128 SecretStore::readKeyContent(std::string const& _content, std::string const&
 	return h128();
 }
 
-bool SecretStore::recode(h128 const& _uuid, string const& _newPass, std::function<std::string()> const& _pass, KDF _kdf)
+bool SecretStore::recode(h128 const& _uuid, string const& _newPass, function<string()> const& _pass, KDF _kdf)
 {
-//	cdebug << "recode:" << toUUID(_uuid);
 	bytes s = secret(_uuid, _pass, true);
 	if (s.empty())
 		return false;
-	m_keys[_uuid].first = encrypt(s, _newPass, _kdf);
+	m_cached.erase(_uuid);
+	m_keys[_uuid].encryptedKey = encrypt(s, _newPass, _kdf);
```

Comment:
The exceptions that are thrown there are actually kind of failed assertions, they should not really happen. What is more likely to happen are io errors and they are also not caught here.
So basically, this function returns false if the password was wrong and throws an exception in exceptional circumstances.